openpiton/bootrom/linux/src/sd.c

```diff
@@ -22,7 +22,18 @@ int sd_copy(void *dst, uint32_t src_lba, uint32_t size)
 
     uint64_t * addr = (uint64_t *)raw_addr;
     volatile uint64_t * p = (uint64_t *)dst;
+    int old = -1;
     for (uint32_t blk = 0; blk < size; blk++) {
+        if(((int)blk)/0x100 > old) {
```

Comment:
this might be clearer if it is solved with a modulo
```suggestion
        if(blk % 100 == 0) {
```